### DISCUSSION TITLE
How your guys create and manage terminals in lazyvim?

### DISCUSSION THREAD

**Discussion by LintaoAmons** [LazyVim/LazyVim#1713] [Q&A]
Can you share your usecase with me?

My current workflow is not very fluent.

I use edge and use `<C-\>` to toggle terminal inside nvim.
![image](https://github.com/LazyVim/LazyVim/assets/95092244/e7af551f-20ed-4816-96b8-bab45f12a13c)

And if I want to have more terminal, I use `<M-3>` to toggle tmux popup window and create and split tmux panes to have more terminals.
![image](https://github.com/LazyVim/LazyVim/assets/95092244/65488686-07b9-4fc7-88ce-6ab828a86d76)

But I was thinking maybe I could just split the nvim terminal buffer and switch between nvim terminals and maybe let some of the terminal run in background or kill current terminal.

**Comment by dpetka2001** [comment]
You can use [toggleterm](https://github.com/akinsho/toggleterm.nvim). It supports creating terminals with `count` prefix. See my [dotfiles](https://github.com/dpetka2001/dotfiles/blob/main/dot_config/nvim/lua/plugins/toggleterm.lua) for example. I can create multiple floating terminals with `1<leader>Tf`, `2<leader>Tf` and so on. You can give names to the terminals with `ToggleTermSetName` and select from multiple terminals with `TermSelect` or `1<C-\>`, `2<C-\>` if you remember what you do in each terminal that you have created with `count`. But everyone has his own preferences regarding his workflow, so in the end `tmux` might be more convenient for you.

**Reply by dpetka2001** [reply]
You can even set new terminals to open for specific things you want to do. See `toggleterm` author's [dotfiles](https://github.com/akinsho/dotfiles/blob/main/.config/nvim/lua/as/plugins/terminal.lua) for inspiration.

**Reply by folke** [reply]
fyi, the lazy terminal also supports count

**Reply by salvaft** [reply]
Hello,
I tried looking for lazyterm but havent found any docs online. <space>ft keybinding calls `Util.lazyterm`
@folke Is there any way to customize lazyterm to get it integrated and showing instead of floating?
thanks

**Reply by folke** [reply]
@SFToro yep, just add the `edgy` extra

**Reply by LintaoAmons** [reply]
Hi @folke ~ 
Can you advise me on how to create a new terminal when I split the terminal window? Alternatively, should I ask how to create a new terminal window that remains at the bottom and can be hidden or maximized using shortcuts?

As shown in the following picture, when I split the terminal window, I expect it to create a new terminal instead of displaying the same terminal in two windows.
![show](https://github.com/LazyVim/LazyVim/assets/95092244/1c8c2680-c6e0-49a3-9c5d-1f06e79d3ab9)

**Reply by folke** [reply]
Use a count, so:

- `<leader>/` is terminal 1, so the same as `1<leader>/`
- `2<leader>/` will toggle terminal 2
- and so on

**Reply by salvaft** [reply]
Hi again...
I see in the edgy repo https://github.com/folke/edgy.nvim that animations can be disabled:
edgy.nvim comes with the following defaults:

```lua
{
 --[...]
  -- edgebar animations
  animate = {
enabled = true
--[...]
```

Id like to disable the sliding neo tree git status box.
I tried:
```lua
return {
  "folke/edgy.nvim",
  opts = function(_, opts)
    opts.animate.enabled = false
  end,
}
```
But animate is undefined.
Thanks in advance

**Reply by folke** [reply]
then just define it?

`opts.animate = {enabled = false}`

**Reply by vdegenne** [reply]
> Use a count, so:
> 
> * `<leader>/` is terminal 1, so the same as `1<leader>/`
> * `2<leader>/` will toggle terminal 2
> * and so on

I don't have edgy installed and this behavior works for me, how commes?

**Reply by folke** [reply]
That's new since the snacks.nvim integration

**Reply by vdegenne** [reply]
Love it!

**Comment by dadav** [comment]
I use toggleterm which is bind to alt+t.
But I rarely use it, because I'm used to just switch the window (i3/sway) and pop a terminal.

**Comment by bijuli74** [comment]
Hi can anyone help me to unbind ```<c-/> & <C-/>``` so that I can bind ```<C-/> ``` to toggle commenting 'gcc'.  In my keymaps.lua file 
```lua 
local Util = require("lazyvim.util")
local lazyterm = function()
  Util.terminal(nil, { cwd = Util.root() })
end

local keymap = vim.keymap
local opts = { noremap = true, silent = true }

-- Select all
keymap.set("n", "<C-a>", "gg<S-v>G")

-- Terminal
keymap.set("n", "<m-;>", lazyterm, { desc = "Terminal (root dir)" })
keymap.set("t", "<M-;>", "<cmd>close<cr>", { desc = "Hide Terminal" })

-- Comments
vim.api.nvim_del_keymap("n", "<c-/>")
vim.api.nvim_del_keymap("t", "<C-/>")

vim.api.nvim_set_keymap(
  "n",
  "<c-/>",
  ':lua require("mini.comment").toggle_lines(vim.api.nvim_win_get_cursor(0)[1], vim.api.nvim_win_get_cursor(0)[1])<CR>',
  opts
)
```

```<C-/> is still opening the terminal along with the new <M-;>. ```
Plz help i am new to neovim, I tried for a long time to no avail. Thanks

**Reply by dpetka2001** [reply]
Also delete `<C-_>`, because some terminals interpret them the same.

**Comment by bijuli74** [comment]
I managed to set <C-/> to 'gcc' finally and <Alt-/> to gc. Here's the code any one need it
```lua -- Keymaps are automatically loaded on the VeryLazy event
-- Default keymaps that are always set: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/keymaps.lua
-- Add any additional keymaps here

local Util = require("lazyvim.util")
local lazyterm = function()
  Util.terminal(nil, { cwd = Util.root() })
end

local keymap = vim.keymap
local opts = { noremap = true, silent = true }

-- Select all
keymap.set("n", "<C-a>", "gg<S-v>G")

-- Comments
vim.api.nvim_del_keymap("n", "<c-/>")
vim.api.nvim_del_keymap("t", "<C-/>")
vim.api.nvim_del_keymap("n", "<c-_>")
vim.api.nvim_del_keymap("t", "<C-_>")

-- Toggle Comment current line
vim.api.nvim_set_keymap(
  "n",
  "<C-/>",
  ':lua require("mini.comment").toggle_lines(vim.api.nvim_win_get_cursor(0)[1], vim.api.nvim_win_get_cursor(0)[1])<CR>',
  opts
)
vim.api.nvim_set_keymap(
  "n",
  "<C-_>",
  ':lua require("mini.comment").toggle_lines(vim.api.nvim_win_get_cursor(0)[1], vim.api.nvim_win_get_cursor(0)[1])<CR>',
  opts
)

-- Toggle Comment selected text
vim.api.nvim_set_keymap(
  "v",
  "<M-/>",
  ':lua require("mini.comment").toggle_lines(vim.fn.line("\'<"), vim.fn.line("\'>"))<CR>',
  opts
)

-- Floating Terminal
keymap.set("n", "<m-;>", lazyterm, { desc = "Terminal (root dir)" })
keymap.set("t", "<M-;>", "<cmd>close<cr>", { desc = "Hide Terminal" })
```

**Reply by vdegenne** [reply]
*(I see you've bind C-a for select all but `C-a` is natively bind for incrementing values, how do you manage them with this config?)*

**Comment by 1himan** [comment]
how to view all currently open/running terminals?

**Reply by dpetka2001** [reply]
Not possible without writing some custom code using `Snacks.terminal` API. You might want to use [toggleterm](https://github.com/akinsho/toggleterm.nvim) for that. Or read [Snacks.terminal](https://github.com/folke/snacks.nvim/blob/main/lua/snacks/terminal.lua) code and implement it yourself.
